### PR TITLE
Display memory usage as factor of 1024 rather than 1000

### PR DESCRIPTION
**Pull request by BasilHorowt**
I think this is expectation for memory usage and makes the numbers line up with what the OS reports

### CODE REVIEW

**Comment by BasilHorowt** [comment]
Potentially closes part of https://github.com/ericaltendorf/plotman/issues/228

**Comment by BasilHorowt** [comment]
Apologies, I am not able to run `pytest` ("fixture 'mocker' not found"), might not have nailed the tests

**Comment by altendky** [comment]
`myenv/bin/pip install --editable .[dev]` should get the deps.

File: src/plotman/plot_util.py [hunk 1]
```diff
@@ -12,11 +12,14 @@ def df_b(d):
 def get_k32_plotsize():
     return 108 * GB
 
-def human_format(num, precision):
+def human_format(num, precision, powerOfTwo=False):
     magnitude = 0
     while abs(num) >= 1000:
```
Comment (altendky) [inline src/plotman/plot_util.py:17]: This needs to vary as well.  Maybe start with `divisor = 1024 if power_of_two else 1000`.

**Review by altendky — CHANGES_REQUESTED** [review]
@ericaltendorf opinions?  I wonder about adding the full `GB` and `GiB`.  `df` uses `T` and such and I don't know which it means or if it is 'universal' or just presumed due to being a disk utility...  `free` does use `Gi`, so maybe that's the 'short' way.  `G` and `Gi`.  Could just add an `i` if `power_of_two`.

**Comment by altendky** [comment]
Oops, said not ready for review...  Well, it seemed close enough to be worth reviewing I guess.  `:]`

**Comment by BasilHorowt** [comment]
> Oops, said not ready for review... Well, it seemed close enough to be worth reviewing I guess. `:]`

Oh sorry! I am ready for review... just not confidant in any way in my python so I guess I was using "Draft" as ... "This might be a decent way to do it". Thank you for the review!

File: src/plotman/plot_util.py (hunk 1 above)
Comment (BasilHorowt) [reply src/plotman/plot_util.py:17]: ah yes, will update

Comment (BasilHorowt) [reply src/plotman/plot_util.py:17]: ah yes, will update

**Comment by BasilHorowt** [comment]
>myenv/bin/pip install --editable .[dev] should get the deps.

@altendky 

Thank you, you mentioned that on keybase too but i never got back to you, basically i get a weird error:

> pip install --editable .[dev]
zsh: no matches found: .[dev]

However I installed the [dev deps ](https://github.com/ericaltendorf/plotman/blob/5788e7fa2df2186ccb94db2d388766f9d52116a4/setup.cfg#L58) manually and now `pytest` works

File: src/plotman/plot_util.py [hunk 2]
```diff
@@ -12,11 +12,13 @@ def df_b(d):
...
+    divisor = 1024 if powerOfTwo else 1000
+    
     magnitude = 0
-    while abs(num) >= 1000:
+    while abs(num) >= divisor:
         magnitude += 1
-        num /= 1000.0
+        num /= divisor        
     return (('%.' + str(precision) + 'f%s') %
             (num, ['', 'K', 'M', 'G', 'T', 'P'][magnitude]))
```
Comment (altendky) [inline src/plotman/plot_util.py:23]: Sorry this got left hanging...  Does this seem ok to you?  Does seem like it should be indicated.

```suggestion
    result = (('%.' + str(precision) + 'f%s') %
            (num, ['', 'K', 'M', 'G', 'T', 'P'][magnitude]))

    if powerOfTwo and magnitude > 0:
	    result += 'i'
    
    return result
```

File: src/plotman/_tests/plot_util_test.py [hunk 3]
```diff
@@ -11,6 +11,8 @@ def test_human_format():
     assert (plot_util.human_format(3542000, 2) == '3.54M')
     assert (plot_util.human_format(354, 0) == '354')
     assert (plot_util.human_format(354, 2) == '354.00')
+    assert (plot_util.human_format(422399296143, 2) == '422.40G')
+    assert (plot_util.human_format(422399296143, 2, True) == '393.39G')
```
Comment (altendky) [inline src/plotman/_tests/plot_util_test.py:15]:
```suggestion
    assert (plot_util.human_format(422399296143, 2, True) == '393.39Gi')
```

File: src/plotman/_tests/plot_util_test.py [hunk 4]
```diff
@@ -11,6 +11,8 @@ def test_human_format():
     assert (plot_util.human_format(3542000, 2) == '3.54M')
     assert (plot_util.human_format(354, 0) == '354')
```
Comment (altendky) [inline src/plotman/_tests/plot_util_test.py:12]:
```suggestion
    assert (plot_util.human_format(354, 0) == '354')
    assert (plot_util.human_format(354, 0, True) == '354')
```

**Review by altendky — CHANGES_REQUESTED** [review]

**Review by ericaltendorf — APPROVED** [review]

**Comment by altendky** [comment]
I got the go ahead from Eric, so once we have the `i` discussion done this is good to go.  Thanks as always.

**Pull request merged**